### PR TITLE
Fix asyncio.CancelledError handling in Sender._fail_all()

### LINKED ISSUE #710: CancelledError exception is not handled inside Sender._fail_all()
**Describe the bug**

When doing a clean shutdown of an app using `asyncio.run(my_app())`, aiofakfa is sometimes failing to handle `asyncio.exceptions.CancelledError`. Here is the example:

```
^C2021-01-19 01:04:31,629 - asyncio(_cancel_all_tasks:61) - ERROR: Exception in callback <bound method Sender._fail_all of <aiokafka.producer.sender.Sender object at 0x7f0ec6542b20>>
handle: <Handle Sender._fail_all>
Traceback (most recent call last):
  File "/home/vvk/miniconda3/envs/fetcher/lib/python3.8/asyncio/runners.py", line 43, in run
    return loop.run_until_complete(main)
  File "uvloop/loop.pyx", line 1450, in uvloop.loop.Loop.run_until_complete
  File "uvloop/loop.pyx", line 1443, in uvloop.loop.Loop.run_until_complete
  File "uvloop/loop.pyx", line 1351, in uvloop.loop.Loop.run_forever
  File "uvloop/loop.pyx", line 519, in uvloop.loop.Loop._run
  File "uvloop/loop.pyx", line 436, in uvloop.loop.Loop._on_idle
  File "uvloop/cbhandles.pyx", line 90, in uvloop.loop.Handle._run
  File "uvloop/cbhandles.pyx", line 70, in uvloop.loop.Handle._run
  File "./bin/order_book_fetcher.py", line 87, in fetch_ws_messages
    await fetcher.log_messages()
  File "/home/vvk/devel/hummingbot-team/exchange-data-fetcher/fetcher/ws_fetcher.py", line 123, in log_messages
    await producer.send(topic_name, item, timestamp_ms=timestamp)
  File "/home/vvk/miniconda3/envs/fetcher/lib/python3.8/site-packages/aiokafka/producer/producer.py", line 439, in send
    partition = self._partition(topic, partition, key, value,
  File "/home/vvk/miniconda3/envs/fetcher/lib/python3.8/site-packages/aiokafka/producer/producer.py", line 366, in _partition
    return self._partitioner(
  File "/home/vvk/miniconda3/envs/fetcher/lib/python3.8/site-packages/kafka/partitioner/default.py", line 15, in __call__
    @classmethod
KeyboardInterrupt

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "uvloop/cbhandles.pyx", line 70, in uvloop.loop.Handle._run
  File "/home/vvk/miniconda3/envs/fetcher/lib/python3.8/site-packages/aiokafka/producer/sender.py", line 62, in _fail_all
    if task.exception() is not None:
asyncio.exceptions.CancelledError
2021-01-19 01:04:40,965 - asyncio(__del__:282) - ERROR: Unclosed AIOKafkaProducer
producer: <aiokafka.producer.producer.AIOKafkaProducer object at 0x7f0ec65429d0>
```

After looking into aiokafka code I found that logic in `aiokafka.producer.Sender._fail_all()` does not account for raised exceptions here:

```python
    def _fail_all(self, task):
        """ Called when sender fails. Will fail all pending batches, as they
        will never be delivered as well as fail transaction
        """
        if task.exception() is not None:  #  <---- here
            self._message_accumulator.fail_all(task.exception())
            if self._txn_manager is not None:
                self._txn_manager.fatal_error(task.exception())
```
 According to https://docs.python.org/3.9/library/asyncio-task.html#asyncio.Task.exception `exception()` method raises `CancelledError` or `InvalidStateError`.

So basically to fix the error, `CancelledError` exception handling should be added into `_fail_all()`.

### PR DESCRIPTION
**Pull request by bitphage**
### Changes

Fixes #710

### Checklist

- [x] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` (e.g. `588.bugfix`)
  * if you don't have an `issue_id` change it to the pr id after creating the PR
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: `Fix issue with non-ascii contents in doctest text files.`

### CODE REVIEW

**Comment by QuentinBrosse** [comment]
Hi @bitphage, and thanks for this fix!

It would be very useful to me. Do you need some help to finish this PR?
It seems like we just need to increase the test coverage to pass all the checks.

File: aiokafka/producer/sender.py [hunk 1]
```diff
@@ -59,10 +59,17 @@ def _fail_all(self, task):
...
         will never be delivered as well as fail transaction
         """
-        if task.exception() is not None:
-            self._message_accumulator.fail_all(task.exception())
+        try:
+            task_exception = task.exception()
+        except asyncio.CancelledError:
+            # Skip setting CancelledError on Futures. Cleanup is already done in
+            # _sender_routine()
+            return
```
Comment (ods) [inline aiokafka/producer/sender.py:67]:
```suggestion
        if task.cancelled():
            return
        task_exception = task.exception()
```

**Comment by ods** [comment]
> It seems like we just need to increase the test coverage to pass all the checks.

Yes, the test case for it would be very welcome

**Comment by QuentinBrosse** [comment]
@ods So, as I'm not allowed to edit the @bitphage branch, I created [a new one on my fork](https://github.com/aio-libs/aiokafka/compare/master...QuentinBrosse:fix/sender-fail_all-cancelled-error) with:
- The fix proposed by @bitphage (edited a bit to address your comments and change as few lines as possible)
- Add a test case

If the changes look good to you, I don't know if you prefer to update the base branch of this PR or close this one and opened a new one based on my branch? :)

**Comment by ods** [comment]
@QuentinBrosse The fix looks good to me. But the test case doesn't check the problem we trying to fix. And it succeeds both for code with and without this fix applied.

**Comment by QuentinBrosse** [comment]
@ods Oups, sorry for that. I'm a very new user of this project so it's not simple to understand everything at the first time.
The test coverage context said that [the return line](https://github.com/aio-libs/aiokafka/compare/master...QuentinBrosse:fix/sender-fail_all-cancelled-error#diff-c06348b175694ff0e0a0f0f22dedb9e73ff8d7325ddc20a9439a5954962fdc0eR63) is covered. So, for me it was ok.

Can you give me some hints on how to improve this test, please?

**Pull request merged**